### PR TITLE
Filter silenced client events

### PR DESCRIPTION
**Pull request by rmc3**
This PR has two components, related to https://github.com/sensu/uchiwa/issues/264

1. Adds a filter, ```hideClientSilenced```, which will hide events from the Event view if the event's ```client.acknowleged``` property is ```true```. Also adds a "Silenced Clients" option to the Hide drop down menu which toggles this filter.
2. Modifies the event view such that the icons in the second column silence/unsilence/show the silence status of the event source (client) and adds a column after the third column containing icons that silence/unsilence/show the silence status of the check.

This is dependent on a forthcoming PR to sensu/uchiwa which will add the client.acknowleged property to events.

### CODE REVIEW

**Comment by rmc3** [comment]
The dependent PR is https://github.com/sensu/uchiwa/pull/268

File: partials/views/events.html [hunk 1]
```diff
@@ -46,14 +47,17 @@
               </tr>
             </thead>
             <tbody>
-              <tr ng-repeat="event in events | orderBy:predicate:reverse | filter:filters.q | filter:{dc:filters.dc} | hideSilenced:filters.silenced | hideOccurrences:filters.occurrences | limitTo:filters.limit | buildEvents track by event._id" ng-click="go('/client/'+event.dc+'/'+event.client.name+'?check='+event.check.name)">
+              <tr ng-repeat="event in events | orderBy:predicate:reverse | filter:filters.q | filter:{dc:filters.dc} | hideSilenced:filters.silenced | hideClientSilenced:filters.clientSilenced | hideOccurrences:filters.occurren:es | limitTo:filters.limit | buildEvents track by event._id" ng-click="go('/client/'+event.dc+'/'+event.client.name+'?check='+event.check.name)">
```
Comment (palourde) [inline partials/views/events.html]: `hideOccurrences:filters.occurren:es` 

Typo?

Comment (rmc3) [inline partials/views/events.html]: Good catch, apologies. Fixed in ba463d8.

**Comment by palourde** [comment]
Sorry for the delay, it has been a busy week at work.

I like the hideClientSilenced filter, great work!

Concerning the additional speaker icon, I feel like it might be confusing. Also, I'm not convinced that we need the icon for the client status in the events view, if we have the hideClientSilenced filter. Any thoughts @amdprophet ?

**Comment by rmc3** [comment]
I definitely understand how having two icons could be seen as confusing, particularly since the function of the icons in the second column of the Events table changes but the appearance does not. 

I see them as a benefit as they provide better visibility into the suppression status of an event. In one view, you can now definitively tell whether an event has been suppressed by a stash, and what the source of that stash is. It also makes it easier to see which events will be hidden when when toggling the Silenced Clients filter.

There's also precedent for this in sensu-dashboard, which I see as a spiritual ancestor of Uchiwa. However, that also makes me biased -- because I used sensu-dashboard and worked on another dashboard that displayed Sensu data with a similar design in the Events view, it makes perfect sense to me, when it might not to someone who has never used those projects.

I'm happy to remove that functionality if there's consensus that it's too confusing to be worthwhile.

**Comment by rmc3** [comment]
For visibility, fixed merge conflict preserving ```ng-if="user.canPost()"``` on the stash() buttons.

**Comment by amdprophet** [comment]
@palourde @rmc3 I like the idea of additional visibility by showing that a client is silenced on the events page. I have no opinions on how it should be shown.

**Comment by rmc3** [comment]
@amdprophet @palourde If there are ideas on how we might indicate that the client is silenced on the events page in a less confusing manner, I'm happy to work on implementing them.

**Comment by amdprophet** [comment]
@rmc3 can you attach a screenshot of what it currently looks like?

My thoughts are that the speaker icons should be grouped somehow with the check & client names. I'm not quite sure of how to illustrate it other than putting both in a div that has a border around it. Maybe we can just get this functionality merged in and worry about styling in a separate PR.

**Comment by rmc3** [comment]
@amdprophet 

screen shot 2015-02-20 at 12 54 22 pm

names sloppily obscured to protect the innocent

**Comment by amdprophet** [comment]
@rmc3 @palourde I can see the possibility for confusion with the speaker icons not being directly under the `Source` & `Check` columns. Anyone who has used Sensu Dashboard in the past should feel right at home though.

@palourde do you have any ideas for stylistic improvements?

**Comment by rmc3** [comment]
screen shot 2015-02-20 at 1 21 46 pm

@amdprophet @palourde how about this? I'm not terribly great with HTML, probably some clean way to adjust for the difference in width between fa-volume-on and fa-volume-off, but it's a stab at what @amdprophet was talking about.

**Comment by amdprophet** [comment]
@rmc3 apply the fa-fw class to the icons - it will set a fixed width.

**Comment by rmc3** [comment]
screen shot 2015-02-20 at 1 34 22 pm

@amdprophet @palourde with fa-fw on the icon spans. Most recent commit contains this update.

**Comment by rmc3** [comment]
Sorry about the second merge -- forgot to ```git pull``` before implementing that change :)

**Comment by palourde** [comment]
Thanks @rmc3 for the PR!

I like the few adjustments you made, much more cleaner. I'll worry about the stylistic improvements later!

**Pull request merged**